tests/integration_tests.rs

```diff
@@ -2213,12 +2256,49 @@ fn lessopen_uses_shell() {
 #[cfg(unix)]
 #[cfg(feature = "lessopen")]
 #[test]
-fn do_not_use_lessopen() {
+fn do_not_use_lessopen_if_overridden() {
     bat()
         .env("LESSOPEN", "|echo File is %s")
+        .arg("--lessopen")
         .arg("--no-lessopen")
         .arg("test.txt")
         .assert()
         .success()
         .stdout("hello world\n");
 }
+
+#[cfg(unix)]
+#[cfg(feature = "lessopen")]
+#[test]
+fn lessopen_validity() {
+    bat()
+        .env("LESSOPEN", "|echo File is test.txt")
+        .arg("--lessopen")
+        .arg("test.txt")
+        .assert()
+        .success()
+        .stdout("hello world\n")
+        .stderr(
+            "\u{1b}[33m[bat warning]\u{1b}[0m: LESSOPEN ignored: must contain exactly one %s\n",
+        );
+
+    bat()
+        .env("LESSOPEN", "|echo File is %s")
```

Comment:
I am completely unfamiliar LESSOPEN and with %s. Can you link to a good online resource where I can learn more please?

Reply:
I would recommend the [`INPUT PREPROCESSOR` section of the `less(1)` man page](https://www.man7.org/linux/man-pages/man1/less.1.html#INPUT_PREPROCESSOR). It is what I used to implement `$LESSOPEN` support.